src/app/editor/contextualListener.js

```diff
@@ -94,6 +91,27 @@ class ContextualListener {
     }
   }
 
+  _getGasEstimation (results) {
+    this.contractName = Object.keys(results.data.contracts[results.source.target])[0]
+    this.target = results.data.contracts[results.source.target]
+    this.functions = Object.keys(this.target[this.contractName].evm.gasEstimates.external)
+  }
+
+  gasEstimation (node) {
+    if (node.name === 'FunctionDefinition') {
+      var functionName = node.attributes.name
+      var fn
+      for (var x in this.functions) {
+        if (!!~this.functions[x].indexOf(functionName)) {
+          fn = this.functions[x]
+          break
+        }
+      }
+      var estimation = this.target[this.contractName].evm.gasEstimates.external[fn]
```

Comment:
@axic, I got estimation like this, not sure if this is correct, because for some functions it says infinite and for some there is a number, but for a constructor there was nothing. So I thought I have to retreive that one from somewhere else... Any thoughts?

Reply:
It can be a number or the `infinite`, both are correct. 

This is the structure returned by the compiler (not sure about remix internals now):
```
          // Function gas estimates
          gasEstimates: {
            creation: {
              codeDepositCost: "420000",
              executionCost: "infinite",
              totalCost: "infinite"
            },
            external: {
              "delegate(address)": "25000"
            },
            internal: {
              "heavyLifting()": "infinite"
            }
          }
```

For the constructor the `creation.totalCost` should be used.

Also make sure that external functions are listed in `external`, while internal ones in `internal`.

Reply:
Nice, will add it tomorrow. Thanks!

Reply:
@ninabreznik is this possible to add
`codeDepositCost: <value> executionCost: <value>` in the title (in case of constructor)? or even on the main view (as you feel like). As we have the information it would be nice to show it.